### PR TITLE
enh(conf/broker) new output 'unified_sql'

### PR DESCRIPTION
**Pull request by a-launois**
## Description

Create new broker configuration output "Unified SQL"
This will replace existing "Perfdata storage" and "SQL" (signaled as deprecated for now)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [X] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [X] 22.04.x (master)

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/install/php/Update-22.04.0-beta.1.php [hunk 1]
```diff
@@ -135,6 +135,61 @@
         ADD `login_attempts` INT(11) UNSIGNED DEFAULT NULL,
         ADD `blocking_time` BIGINT(20) UNSIGNED DEFAULT NULL"
     );
+
+    /**
+     * Add new UnifiedSQl broker output
+     */
+    $errorMessage = 'Unable to update cb_type';
+
```
Comment (kduret) [inline www/install/php/Update-22.04.0-beta.1.php:143]: you should start transaction here

File: www/install/php/Update-22.04.0-beta.1.php [hunk 2]
```diff
@@ -135,6 +135,61 @@
...
+    $statement = $pearDB->query(
+        "UPDATE `cb_type` set type_name = 'Broker SQL database - DEPRECATED'
+            WHERE type_shortname = 'sql'"
+    );
+
+    $statement = $pearDB->query(
+        "SELECT cb_module_id FROM cb_module
+            WHERE name = 'Storage'"
+    );
+    $moduleId = $statement->fetch();
```
Comment (kduret) [inline www/install/php/Update-22.04.0-beta.1.php:157]: even if it should produce, you need to manage the case where fetch returns false

File: www/install/php/Update-22.04.0-beta.1.php [hunk 3]
```diff
@@ -135,6 +135,61 @@
...
+            VALUES ('Unified SQL', 'unified_sql', $moduleId)"
+    );
+    $typeId = $pearDB->lastInsertId();
+
+    $errorMessage = 'Unable to update cb_tag_type_relation';
+    $statement = $pearDB->query(
+        "SELECT cb_module_id FROM cb_module
+            WHERE name = 'Storage'"
+    );
+    $tagId = $statement->fetch();
```
Comment (kduret) [inline www/install/php/Update-22.04.0-beta.1.php:169]:
```suggestion
    $tagId = $statement->fetch();even if it should produce, you need to manage the case where fetch returns false
```

File: www/install/php/Update-22.04.0-beta.1.php [hunk 4]
```diff
@@ -135,6 +135,75 @@
...
+        throw new Exception("Cannot find fields in cb_type_field_relation table");
+    }
+
+    $query = "INSERT INTO `cb_type_field_relation` (`cb_type_id`, `cb_field_id`, `is_required`, `order_display`)";
+    foreach ($inputs as $key => $input) {
+        $order = $key + 1;
+        $query .= $key === 0 ? " VALUES " : ", ";
+        $query .= "($typeId, " . $input['cb_field_id'] . ", " . $input['is_required'] . ", $order)";
+    }
+    $statement = $pearDB->query($query);
```
Comment (kduret) [inline www/install/php/Update-22.04.0-beta.1.php:204]:
```suggestion
    $pearDB->query($query);
```

**Review by kduret — APPROVED** [review]

**Review by AssanDiomande — APPROVED** [review]

**Pull request merged**